version: "3"
services:
  postgres-server:
    image: postgres:14.0
    container_name: postgres-server
    restart: always
    ports:
      - "5432:5432"
    environment:
      POSTGRES_USER: ${MLOPS_USER}
      POSTGRES_PASSWORD: ${MLOPS_PASSWORD}
      POSTGRES_DB: ${MLOPS_DATABASE}
    networks:
      - mlops_network
    healthcheck:
      test:
        [
          "CMD",
          "pg_isready",
          "-q",
          "-U",
          "${MLOPS_USER}",
          "-d",
          "${MLOPS_DATABASE}",
        ]
      interval: 10s
      timeout: 5s
      retries: 5
  data-generator:
    container_name: data-generator
    build:
      context: .
      dockerfile: Dockerfile-data-generator
    environment:
      MLOPS_USER: ${MLOPS_USER}
      MLOPS_PASSWORD: ${MLOPS_PASSWORD}
      MLOPS_HOST: ${MLOPS_HOST}
      MLOPS_PORT: ${MLOPS_PORT}
      MLOPS_DATABASE: ${MLOPS_DATABASE}
    networks:
      - mlops_network
    depends_on:
      postgres-server:
        condition: service_healthy
    command: ["postgres-server"]

networks:
  mlops_network:
    driver: bridge
